gobblin-core/src/main/java/gobblin/writer/http/AsyncHttpWriterBuilder.java

```diff
@@ -44,11 +52,12 @@
  * @param <RQ> type of request
  * @param <RP> type of response
  */
+@Slf4j
 public abstract class AsyncHttpWriterBuilder<D, RQ, RP> extends FluentDataWriterBuilder<Void, D, AsyncHttpWriterBuilder<D, RQ, RP>> {
   public static final String CONF_PREFIX = "gobblin.writer.http.";
 
   @Getter
-  protected State state;
+  protected WorkUnitState state;
```

Comment:
@zxcware this abstraction is wrong unfortunately. In a Gobblin execution the builder always receives a `WorkUnitState`. Furthermore, this is not creating a broker, it is getting the already created task level broker. Creating a new one does not produce the desired results.